src/utils.ts

```diff
@@ -50,31 +50,85 @@ export function isFlutterDriverCommand(command: string) {
   );
 }
 
-export function fetchFlutterServerPort(
-  deviceLogs: [{ message: string }],
-): number {
-  let port: number | undefined;
-  for (const line of deviceLogs.map((e) => e.message).reverse()) {
-    const match = line.match(FLUTTER_SERVER_START_MESSAGE);
-    if (match) {
-      log.info(`Found the server started log from device: ${line}`);
-      port = Number(match[1]);
-      break;
-    }
-  }
-  if (!port) {
-    throw new Error(
-      `Flutter server started message '${FLUTTER_SERVER_START_MESSAGE}' was not found in the device log. ` +
-        `Please make sure the application under test is configured properly according to ` +
-        `https://github.com/AppiumTestDistribution/appium-flutter-server and that it does not crash on startup.
-        Also make sure "appium:skipLogcatCapture" is not set to true in the desired capabilities. `,
-    );
-  }
-  log.info(`Extracted the port from the device logs: ${port}`);
-  return port;
-}
-
 export async function getFreePort() {
-  const [start, end] = DEVICE_PORT_RANGE;
+  const [start, end] = SYSTEM_PORT_RANGE;
   return await findAPortNotInUse(start, end);
 }
+
+async function waitForFlutterServer(port: number, packageName: string) {
+  const proxy = new JWProxy({
+    server: '127.0.0.1',
+    port: port,
+  });
+  await waitForCondition(
+    async () => {
+      try {
+        const response: any = await proxy.command('/status', 'GET');
+        if (!response) {
+          log.info(`FlutterServer not reachable on port ${port}, Retrying..`);
+          return false;
+        }
+        if (response.appInfo?.packageName === packageName) {
```

Comment:
it would be safer to also make appInfo optional: response?.appInfo?. packageName